packages/ve-hemi-actions/test/actions/wallet/increaseAmount.test.ts

```diff
@@ -0,0 +1,511 @@
+      validParameters.additionalAmount,
+    )
+    vi.mocked(getErc20TokenAllowance).mockResolvedValue(
+      BigInt(validParameters.additionalAmount),
+    )
+    vi.mocked(writeContract).mockResolvedValue(zeroHash)
+    vi.mocked(waitForTransactionReceipt).mockResolvedValue({
+      status: 'success',
+    })
+
+    const { emitter, promise } = increaseAmount(validParameters)
+
+    const preIncreaseAmount = vi.fn()
+    const userSignedIncreaseAmount = vi.fn()
+    const increaseAmountTransactionSucceeded = vi.fn()
+    const increaseAmountSettled = vi.fn()
+
+    emitter.on('pre-increase-amount', preIncreaseAmount)
+    emitter.on('user-signed-increase-amount', userSignedIncreaseAmount)
+    emitter.on(
+      'increase-amount-transaction-succeeded',
+      increaseAmountTransactionSucceeded,
+    )
+    emitter.on('increase-amount-settled', increaseAmountSettled)
+
+    await promise
+
+    expect(preIncreaseAmount).toHaveBeenCalledOnce()
+    expect(userSignedIncreaseAmount).toHaveBeenCalledWith(zeroHash)
+    expect(increaseAmountTransactionSucceeded).toHaveBeenCalled()
+    expect(increaseAmountSettled).toHaveBeenCalledOnce()
+
+    expect(writeContract).toHaveBeenCalledWith(
+      validParameters.walletClient,
+      expect.objectContaining({
+        address: getVeHemiContractAddress(hemiSepolia.id),
+        args: [validParameters.tokenId, validParameters.additionalAmount],
+        functionName: 'increaseAmount',
+      }),
+    )
+  })
+
+  it('should skip approval if allowance is sufficient', async function () {
+    vi.mocked(getErc20TokenBalance).mockResolvedValue(
+      validParameters.additionalAmount,
+    )
+    vi.mocked(getErc20TokenAllowance).mockResolvedValue(
+      validParameters.additionalAmount + BigInt(1),
+    )
+    vi.mocked(writeContract).mockResolvedValue(zeroHash)
+    vi.mocked(waitForTransactionReceipt).mockResolvedValue({
+      status: 'success',
+    })
+
+    const { emitter, promise } = increaseAmount(validParameters)
+
+    const preApprove = vi.fn()
+    const preIncreaseAmount = vi.fn()
+    const increaseAmountSettled = vi.fn()
+
+    emitter.on('pre-approve', preApprove)
+    emitter.on('pre-increase-amount', preIncreaseAmount)
+    emitter.on('increase-amount-settled', increaseAmountSettled)
+
+    await promise
+
+    expect(preApprove).not.toHaveBeenCalled()
+    expect(preIncreaseAmount).toHaveBeenCalledOnce()
+    expect(increaseAmountSettled).toHaveBeenCalledOnce()
+    expect(approveErc20Token).not.toHaveBeenCalled()
+  })
+
+  it('should handle approve transaction failure', async function () {
+    vi.mocked(getErc20TokenBalance).mockResolvedValue(BigInt(1000))
+    vi.mocked(getErc20TokenAllowance).mockResolvedValue(BigInt(0))
+    vi.mocked(approveErc20Token).mockResolvedValue(zeroHash)
+    vi.mocked(waitForTransactionReceipt)
+      .mockResolvedValueOnce({
+        status: 'reverted',
+      })
+      .mockResolvedValue({
+        status: 'success',
+      })
```

Comment:
If the approve tx fails, I think no other tx should take place 🤔 

```suggestion
    vi.mocked(waitForTransactionReceipt)
      .mockResolvedValueOnce({
        status: 'reverted',
      })
```

Reply:
Fixed here 1cc5f78bf8bac889a2129625fdf46b34644c3ca1. Thanks!